packages/fxa-auth-server/lib/routes/account.js

```diff
@@ -1173,70 +1173,44 @@ module.exports = (
             });
         }
 
-        function resetAccountData() {
-          let authSalt;
-          return random
-            .hex(32)
-            .then(hex => {
-              authSalt = hex;
-              password = new Password(authPW, authSalt, config.verifierVersion);
-              return password.verifyHash();
-            })
-            .then(verifyHashData => {
-              verifyHash = verifyHashData;
-
-              return setupKb();
-            })
-            .then(() => {
-              return db.resetAccount(accountResetToken, {
-                authSalt,
-                verifyHash,
-                wrapWrapKb,
-                verifierVersion: password.version,
-              });
-            })
-            .then(() => {
-              // Delete all passwordChangeTokens, passwordForgotTokens and
-              // accountResetTokens associated with this uid
-              return db.resetAccountTokens(accountResetToken.uid);
-            })
-            .then(() => {
-              // Notify the devices that the account has changed.
-              request.app.devices.then(devices =>
-                push.notifyPasswordReset(accountResetToken.uid, devices)
-              );
-
-              return db
-                .account(accountResetToken.uid)
-                .then(accountData => (account = accountData));
-            })
-            .then(() => {
-              return P.all([
-                request.emitMetricsEvent('account.reset', {
-                  uid: account.uid,
-                }),
-                log.notifyAttachedServices('reset', request, {
-                  uid: account.uid,
-                  generation: account.verifierSetAt,
-                }),
-                customs.reset(account.email),
-              ]);
-            });
-        }
-
-        function setupKb() {
+        async function resetAccountData() {
```

Comment:
Per https://github.com/mozilla/fxa/pull/2570#discussion_r326949419, I asyncified this function while I was in here. The logic should be unchanged, but will change slightly in the promised follow-up commit.